contracts/Strategy.sol

```diff
@@ -53,80 +222,624 @@ contract Strategy is BaseStrategy {
+                    utilization < KASHI_MAXIMUM_TARGET_UTILIZATION &&
+                    utilization > KASHI_MINIMUM_TARGET_UTILIZATION &&
+                    highestUtilization < KASHI_MAXIMUM_TARGET_UTILIZATION &&
+                    highestUtilization > KASHI_MINIMUM_TARGET_UTILIZATION)
+            ) {
+                highestInterest = interestPerBlock;
+                highestUtilization = utilization;
+                _highestIndex = i;
+            }
+        }
+    }
+
+    function lowestInterestPairIndex(uint256 minLiquidShares)
+        internal
+        view
+        returns (uint256 _lowestIndex)
+    {
+        _lowestIndex = type(uint256).max; // Max indicate no low APR with liquidity
+        uint256 lowestInterest = type(uint256).max;
+        uint256 lowestUtilization = KASHI_UTILIZATION_PRECISION;
+
+        for (uint256 i = 0; i < kashiPairs.length; i++) {
+            (uint256 interestPerBlock, , ) =
+                kashiPairs[i].kashiPair.accrueInfo();
+
+            uint256 utilization =
+                kashiPairUtilization(kashiPairs[i].kashiPair, 0);
+
+            // A pair is lowest if either
+            //   - It's utilization is lower, and either
+            //     - It is below the min taget util
+            //     - The existing choice is above the max target util
+            //   - Compare APR directly only if both are between the min and max
+            if (
+                ((utilization < lowestUtilization &&
+                    (lowestUtilization > KASHI_MAXIMUM_TARGET_UTILIZATION ||
+                        utilization < KASHI_MINIMUM_TARGET_UTILIZATION)) ||
+                    (interestPerBlock < lowestInterest &&
+                        utilization < KASHI_MAXIMUM_TARGET_UTILIZATION &&
+                        utilization > KASHI_MINIMUM_TARGET_UTILIZATION &&
+                        lowestUtilization < KASHI_MAXIMUM_TARGET_UTILIZATION &&
+                        lowestUtilization >
+                        KASHI_MINIMUM_TARGET_UTILIZATION)) &&
+                kashiFractionTotal(i) > dustThreshold &&
+                kashiPairLiquidShares(i) >= minLiquidShares
+            ) {
+                lowestInterest = interestPerBlock;
+                _lowestIndex = i;
+            }
+        }
+    }
+
+    function kashiPairUtilization(IKashiPair kashiPair, uint256 sharesToDeposit)
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 totalAssetShares = kashiPair.totalAsset().elastic;
+        uint256 totalBorrowAmount = kashiPair.totalBorrow().elastic;
+        uint256 fullAssetAmount =
+            bentoSharesToWant(totalAssetShares.add(sharesToDeposit), false).add(
+                totalBorrowAmount
+            );
+
+        return
+            uint256(totalBorrowAmount).mul(KASHI_UTILIZATION_PRECISION).div(
+                fullAssetAmount
+            );
+    }
+
+    function wantToBentoShares(uint256 wantAmount, bool roundUp)
```

Comment:
i don't see call where `roundUp` argument is false, is this needed? or a setting or hardcoded value works? removing would reduce noise